objection_engine/parse_tags.py

```diff
@@ -0,0 +1,194 @@
+from dataclasses import dataclass
```

Comment:
This is a question out of curiosity, not a petition. Are the tags escapable? Like is there a way to render "<blue>example</blue>" all in black? Like with s backslash (\) or something

I don't really think I need this, this shouldn't be needed but I'm curious nevertheless

Reply:
At the moment, I don't think there is a way to escape tags (at least, I didn't program one in myself). That certainly seems like something we could add with a future pull request, though.

In thinking about this, I realized that I didn't really try to protect against faulty strings; for example, I've no idea what
```
Hello, my name is</red> Phoenix
```
would cause the engine to do. I suppose for now we could just say the input should always be sanitized/correct otherwise the program will crash, but ensuring that it can handle invalid tag order and such would be important.

Reply:
Ok. So, revisiting this one. As you've said, it crashes in those two moments.

- [ ] Closing tag without starting one
- [ ] Mismatched tags

I have found through testing that it also crashes on:

- [ ] Non-existent color.

That being said I think we should avoid these three crashes (and any others). The easiest way would be that if the engine crashes while trying to render rich text just ignore all rich text and render it as plain text.

This is because I don't really think we should pass the responsibility of sanitizing the input to the app using the library. And the end users (where text is generated) are free to write whatever they want

Reply:
That sounds good to me!

I've gone ahead and added code to handle the first two cases - if it runs into one of those, it should just render that block of text without any coloring (the tags will be visible). For the third case (non-existent color), it currently just renders that text as white.

Here's a video of it in action (no sound):

https://user-images.githubusercontent.com/9957987/200202996-6d959e4a-fbfd-4ec9-bf76-23a39b028f26.mp4

Do you think this is the right way to handle the non-existent tag, or should the tag and closing tag be visible as with the other error cases?

Reply:
Both approaches would be acceptable but I think rendering it as white is in fact better